cozy/model/settings.py

```diff
@@ -47,11 +47,11 @@ def last_played_book(self, new_value):
         self._db_object.save(only=self._db_object.dirty_fields)
 
     @property
-    def default_location(self) -> bool:
+    def default_location(self) -> Storage | NoReturn:
         for location in self.storage_locations:
             if location.default:
-                return True
-        return False
+                return location
+        assert False, "this should never happen"
```

Comment:
Note that assertions can be optimized away if Python is started with `-O`.

Reply:
Yeah, but in this case it doesn't really matter. With the current code this assert statement can never be reached, and I only put it there so that if someone breaks it again, they will notice the assertion during development.

Reply:
It is kind of an interesting case of typing in Python I haven't encountered/considered before - the validity of the return type depends on the way you start Python :thinking:

Reply:
I've [asked](https://github.com/python/typing/discussions/1598#discussioncomment-8189484) on [`python/typing`](https://github.com/python/typing) and the suggestion was to `raise AssertionErorr` instead.
```suggestion
        raise AssertionError("This should never happen")
```

Reply:
Cool, thanks!